cmd/template/nodepool/provider/azure.go

```diff
@@ -33,6 +35,10 @@ func WriteAzureTemplate(out io.Writer, config NodePoolCRsConfig) error {
 		infrastructureRef := newCAPZMachinePoolInfraRef(azureMachinePoolCR)
 
 		machinePoolCR := newCAPIV1Alpha3MachinePoolCR(config, infrastructureRef)
+		{
+			machinePoolCR.GetAnnotations()[annotation.NodePoolMinSize] = strconv.Itoa(config.NodesMin)
```

Comment:
Currently we set the `replicas` field of the MachinePool CR to the `max` value.
Should we maybe set it to the `min` value?

```
$ ./kubectl-gs template nodepool --provider azure --cluster-id j7gaw --owner giantswarm --nodes-min=1 --nodes-max=3
...
apiVersion: exp.cluster.x-k8s.io/v1alpha3
kind: MachinePool
metadata:
  annotations:
    cluster.k8s.io/cluster-api-autoscaler-node-group-max-size: "3"
    cluster.k8s.io/cluster-api-autoscaler-node-group-min-size: "1"
    machine-pool.giantswarm.io/name: Unnamed node pool
  ...
spec:
  ...
  replicas: 3
...
```

Reply:
Good catch 👍🏻